tests/units/units.cpp

```diff
@@ -1060,19 +1060,132 @@ TEST(Units, scalingFactorWithTwoEmptyUnits)
     EXPECT_EQ(0.0, libcellml::Units::scalingFactor(u2, u1));
 }
 
-TEST(Units, compareIncompatiableMultiplierSimple)
+// TEST(Units, scalingFactorBetweenMultipliedUnits)
+// {
+//     // u1 = 1000*u2 ... but is it really ... ?
+//     libcellml::UnitsPtr u1 = libcellml::Units::create();
+//     u1->setName("u1");
+//     u1->addUnit("metre", 0, 3.0, 1000.0);
+//     libcellml::UnitsPtr u2 = libcellml::Units::create();
+//     u2->setName("u2");
+//     u2->addUnit("metre", 0, 1.0, 1.0);
+//     u2->addUnit("metre", 0, 1.0, 1.0);
+//     u2->addUnit("metre", 0, 1.0, 1.0);
+
+//     EXPECT_EQ(1000.0, libcellml::Units::scalingFactor(u2, u1));
+//     EXPECT_EQ(0.001, libcellml::Units::scalingFactor(u1, u2));
+// }
+
+TEST(Units, scalingFactorBetweenDissimilarUnits)
+{
+    libcellml::UnitsPtr u1 = libcellml::Units::create();
+    u1->setName("u1");
+    u1->addUnit("metre", 0, 1.0, 1000.0);
+    libcellml::UnitsPtr u2 = libcellml::Units::create();
+    u2->setName("u2");
+    u2->addUnit("metre", 0, 1.0, 1.0);
+    u2->addUnit("metre", 0, 1.0, 1.0);
+    u2->addUnit("metre", 0, 1.0, 1.0);
+
+    EXPECT_EQ(0.0, libcellml::Units::scalingFactor(u2, u1));
+}
+
+TEST(Units, compareIncompatibleMultiplierSimple)
 {
-    // u1 = 1000*u2
     libcellml::UnitsPtr u1 = libcellml::Units::create();
     u1->setName("u1");
     u1->addUnit("metre", 0, 1.0, 1.0);
     libcellml::UnitsPtr u2 = libcellml::Units::create();
     u2->setName("u2");
     u2->addUnit("second", 0, 1.0, 1.0);
 
-    // Incompatible units but return a scaling factor of 1.0.
+    // Incompatible units return a scaling factor of 0.0.
+    EXPECT_EQ(0.0, libcellml::Units::scalingFactor(u1, u2));
+    EXPECT_EQ(0.0, libcellml::Units::scalingFactor(u2, u1));
```

Comment:
Do we really need that second test?

Reply:
Since you've asked for reverse tests to be added below, I'm going to go with "yes" ;)

Reply:
The difference, though, is that here we return `0.0` while in the other case I would expect to see `s` and `1.0/s`. However, I am happy with testing both ways in all cases.